clippy_lints/src/methods/mod.rs

```diff
@@ -2519,6 +2538,15 @@ fn lint_into_iter(cx: &LateContext<'_, '_>, expr: &hir::Expr, self_ref_ty: Ty<'_
     }
 }
 
+fn lint_suspicious_map(cx: &LateContext<'_, '_>, expr: &hir::Expr) {
+    span_lint(
+        cx,
+        SUSPICIOUS_MAP,
+        expr.span,
+        "Make sure you did not confuse `map` with `filter`.",
```

Comment:
Someone should document the rules of rust error messages and suggestions somewhere somtimes. 😄 

```suggestion
        "make sure you did not confuse `map` with `filter`",
```

IMO this should be a `span_help_and_lint`: 
- message: "this call to `map()` won't have an effect on the call to `count()`"
- help: "make sure you did not confuse `map` with `fliter`"